core/internal/api/logging.go

```diff
@@ -49,11 +50,15 @@ func withRetryLogging(
 			case err != nil:
 				logger.Info("api: retrying error", "error", err)
 			case resp.StatusCode >= 400:
+				// TODO: Log the request body.
 				logger.Info(
 					"api: retrying HTTP error",
 					"status", resp.StatusCode,
 					"url", resp.Request.URL.String(),
 				)
+
+				// TODO: Report the attempt number & time to next retry.
```

Comment:
ah nice! i like it... i think the retry package does it in our logs

Reply:
unfortunately there's no way to peek on that information with `retryablehttp`; this could be another reason to replace it with our own solution

Reply:
haha... i know you are not a fan, so if you collect 10 reasons we might just do it 😜